app/components/emoji_keyboard.js

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React from 'react';
+import {
+    View,
+    Text,
+    StyleSheet,
+    FlatList,
+    TouchableOpacity
+} from 'react-native';
+import {KeyboardRegistry} from 'react-native-keyboard-input';
+import {store} from 'app/mattermost';
```

Comment:
Actually it does need it. Since the custom keyboard is a react-native application on it own (separate from the main app), the redux-aware emoji component has no access to the store unless we pass it through props or context. Initially I used a `Provider` but I switched to a more minimalistic way to pass the store. I'll switch back to `Provider` for sake of clarity and because I want to pull some theme features directly from the custom keyboard.

Here's a screenshot with what happens if the emoji component does not have access to the store.

<img width="362" alt="screen shot 2017-06-20 at 7 02 07 pm" src="https://user-images.githubusercontent.com/1720752/27359830-f121d592-55eb-11e7-97b9-f48ef195a51c.png">